collections/guides/use/run-your-own-code-in-a-reactor.md

```diff
@@ -0,0 +1,345 @@
+
+We will allow setting a single configuration parameter with this Reactor, the API key issued by Lithic for our account:
+
+| Name           | Type    | Description                   |
+|:---------------|:--------|:------------------------------|
+| LITHIC_API_KEY | string  | The api key issued by Lithic  |
+
+We will also allow requests to our Reactor to specify several parameters that we will forward to the Lithic API to 
+control the name of our new card, the type of card to create, and the spend limit of the card:
+
+| Name                 | Type   | Description                                                                                                            |
+|:---------------------|:-------|:-----------------------------------------------------------------------------------------------------------------------|
+| memo                 | string | The name to identify the card                                                                                          |
+| type                 | string | The type of card to issue ([see Lithic docs](https://docs.lithic.com/docs/cards#cardtype))                             |
+| spend_limit          | number | The amount (in cents) to limit approved authorizations                                                                 |
+| spend_limit_duration | string | Duration for which spend limit applies ([see Lithic docs](https://docs.lithic.com/docs/cards#cardspend_limit_duration) |
+
+We can then use these configuration and request parameters in our Reactor Formula code to make an API request to Lithic.
+We will also create a `card` token as output from this Reactor using the card details provided to us in the Lithic API response. 
+
+```js
+module.exports = async function (req) {
+  const { LITHIC_API_KEY } = req.configuration;
+  const { memo, type, spend_limit, spend_limit_duration } = req.args;
+
+  const fetch = require('node-fetch');
+
+  const body = {
+    memo,
+    type,
+    spend_limit,
+    spend_limit_duration,
+    state: 'OPEN'
+  };
+
+  const response = await fetch(
+    `https://sandbox.lithic.com/v1/card`,
+    {
+      method: 'post',
+      body: JSON.stringify(body),
+      headers: { 
+        'Authorization': `api-key ${LITHIC_API_KEY}`,
+        'Content-Type': 'application/json',
+      }
+    }
+  );
+  // omitted for brevity: translate response status codes to Basis Theory Reactor errors
+    
+  const lithicCard = await response.json();
+
+  return { 
+    tokenize: {
+      type: 'card',
+      data: {
+        number: lithicCard.pan,
+        expiration_month: lithicCard.exp_month,
+        expiration_year: lithicCard.exp_year,
+        cvc: lithicCard.cvv
+      },
+      metadata: {
+        lithic_token_id: lithicCard.token
+      }
+    }
+  };
+}
+```
+
+Next, we need to create a Private Reactor Formula in Basis Theory containing this `code`, `configuration`, and `request_parameters` by calling the 
```

Comment:
i personally found this step confusing when creating my first custom reactor. having to stringify the function for the `code` property, listing all the args and config values, etc. i think an example of the Lithic formula object might be a good idea to show here?

Reply:
Oh, man... I would really like to add this example, but the curl request is 86 lines long... is it worth taking up that much space?

Reply:
I added this example - I got it to be a bit shorter but it's still quite long and with a lot of side scrolling to side the code string

Reply:
@dhudec i think you could do something like `"code": "module.exports = async function (req) {\n ...` as you have the "Note" above.